common/djangoapps/microsite_configuration/middleware.py

```diff
@@ -83,3 +87,54 @@ def _set_cookie_wrapper(key, value='', max_age=None, expires=None, path='/', dom
             response.set_cookie = _set_cookie_wrapper
 
         return response
+
+
+class DatabaseMicrositeMiddleware(MicrositeMiddleware):
+    """
+    Middleware class which will bind configuration information regarding 'microsites' on a per request basis.
+    The actual configuration information is taken from the microsite model in the database
+    """
+
+    def process_request(self, request):
+        """
+        Middleware entry point on every request processing. This will associate a request's domain name
+        with a 'University' and any corresponding microsite configuration information
+        """
+        microsite.clear()
+        domain = request.META.get('HTTP_HOST', None)
+        microsite.set_from_db_by_domain(domain)
+        return None
+
+
+class MicrositeCrossBrandingFilterMiddleware():
+    """
+    Middleware class that prevents a course defined in a branded ORG trough a microsite, to be displayed
+    on a different microsite with a different branding.
+    """
```

Comment:
Since for us the 'preview' button from studio points the instance without branding, and our customers have some heavy branding, we discourage the use of said button. So the request are indeed blocked, but it is not a big issue for us now.

What could be done here would be to either leave the use of this filter as a per microsite configuration, or to add a whitelist of domains who can render any course.